autobump-bitbucket/src/main/java/model/BitBucketGitProvider.java

```diff
@@ -0,0 +1,47 @@
+package model;
+
+import com.github.autobump.core.model.GitProvider;
+import com.github.autobump.core.model.PullRequest;
+import com.github.autobump.core.model.PullRequestResponse;
+import feign.Feign;
+import feign.auth.BasicAuthRequestInterceptor;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import lombok.Getter;
+import model.dtos.PullRequestBodyDto;
+import model.dtos.PullRequestResponseDto;
+
+@Getter
+public class BitBucketGitProvider implements GitProvider {
+    private final String apiUrl;
+    private final BitBucketAccount user;
+
+    public BitBucketGitProvider(BitBucketAccount user, String apiUrl) {
+        this.user = user;
+        this.apiUrl = apiUrl;
+    }
+
+    @Override
+    public PullRequestResponse makePullRequest(PullRequest pullRequest) {
+        PullRequestBodyDto body = new PullRequestBodyDto(pullRequest.getTitle(),
+                new PullRequestBodyDto.Source(new PullRequestBodyDto.Branch(pullRequest.getBranchName())));
+
+        BitBucketApi client = Feign.builder()
```

Comment:
cant this client be made in the constructor?

Reply:
The Feign client (result of the Feign.builder()...) is indeed an immutable service that can be shared by multiple threads. It also involves some work to create this object because the interface you defined needs to be introspected. Therefor it is best to define the Feign client (BitBucketApi) as a field and re-use it.